controllers/barbican_controller.go

```diff
@@ -555,22 +538,32 @@ func (r *BarbicanReconciler) reconcileDelete(ctx context.Context, instance *barb
 
 // fields to index to reconcile when change
 const (
-	passwordSecretField     = ".spec.secret"
-	caBundleSecretNameField = ".spec.tls.caBundleSecretName"
-	tlsAPIInternalField     = ".spec.tls.api.internal.secretName"
-	tlsAPIPublicField       = ".spec.tls.api.public.secretName"
+	passwordSecretField      = ".spec.secret"
+	caBundleSecretNameField  = ".spec.tls.caBundleSecretName"
+	tlsAPIInternalField      = ".spec.tls.api.internal.secretName"
+	tlsAPIPublicField        = ".spec.tls.api.public.secretName"
+	p11LoginSecretField      = ".spec.pkcs11.loginSecret"
+	p11ClientDataSecretField = ".spec.pkcs11.clientDataSecret"
```

Comment:
there is a bit more todo to index the field, check here for reference https://github.com/openstack-k8s-operators/barbican-operator/blob/main/controllers/barbicanapi_controller.go#L863-L873 . we'd need the same for the new fields in all controllers.
now that we have the job in the barbican controller, we'd also have to add the equivalent to the barbican controller, or move the job to the api controller of that makes sense.